core/sql/optimizer/NAFileSet.cpp

```diff
@@ -161,6 +162,18 @@ Lng32 NAFileSet::getKeyLength()
 	return keyLength_;
 }
 
+// returns the length of the encoded key in bytes for this index
+Lng32 NAFileSet::getEncodedKeyLength()
+{
+	if(encodedKeyLength_ >0) return encodedKeyLength_;
+
+	for(CollIndex i=0;i<indexKeyColumns_.entries();i++)
+	{
+		encodedKeyLength_ += indexKeyColumns_[i]->getType()->getNominalSize();
```

Comment:
What's the reason for not calling the method that already does this, NAType::getEncodedKeyLength()?